tests/common.go

```diff
@@ -291,7 +413,163 @@ func ValidateVolumes(ctx *scheduler.Context, errChan ...*chan error) {
+		})
+
+		var vols map[string]map[string]string
+		Step(fmt.Sprintf("get %s app's volume's custom parameters", ctx.App.Key), func() {
+			vols, err = Inst().S.GetVolumeParameters(ctx)
+			processError(err, errChan...)
+		})
+
+		for vol, params := range vols {
+			if Inst().ConfigMap != "" {
+				params["auth-token"], err = Inst().S.GetTokenFromConfigMap(Inst().ConfigMap)
+				processError(err, errChan...)
+			}
+			if ctx.RefreshStorageEndpoint {
+				params["refresh-endpoint"] = "true"
+			}
+			Step(fmt.Sprintf("get %s app's volume: %s inspected by the volume driver", ctx.App.Key, vol), func() {
+				err = Inst().V.ValidateCreateVolume(vol, params)
+				processError(err, errChan...)
+			})
+			Step(fmt.Sprintf("get %s app's volume: %s then create local snapshot", ctx.App.Key, vol), func() {
+				err = Inst().V.ValidateCreateSnapshot(vol, params)
+				if err != nil {
+					expect(err.Error()).To(contain(errPureFileSnapshotNotSupported.Error()))
+				}
+			})
+			Step(fmt.Sprintf("get %s app's volume: %s then create cloudsnap", ctx.App.Key, vol), func() {
+				err = Inst().V.ValidateCreateCloudsnap(vol, params)
+				if err != nil {
+					expect(err.Error()).To(contain(errPureFileSnapshotNotSupported.Error()))
+				}
+			})
+		}
+	})
+}
+
+// ValidateFBSnapshotsPXCTL is the ginkgo spec for validating FB volume snapshots using PXCTL for a context
+func ValidateFBSnapshotsPXCTL(ctx *scheduler.Context, errChan ...*chan error) {
+	context("For validation of an app's volumes", func() {
+		var err error
+		Step(fmt.Sprintf("inspect %s app's volumes", ctx.App.Key), func() {
+			appScaleFactor := time.Duration(Inst().GlobalScaleFactor)
+			err = Inst().S.ValidateVolumes(ctx, appScaleFactor*defaultTimeout, defaultRetryInterval, nil)
+			processError(err, errChan...)
+		})
+
+		var vols []*volume.Volume
+		Step(fmt.Sprintf("get %s app's pure volumes", ctx.App.Key), func() {
+			vols, err = Inst().S.GetPureVolumes(ctx)
+			processError(err, errChan...)
+		})
+
+		for _, vol:= range vols {
+
+			Step(fmt.Sprintf("get %s app's volume: %s then create snapshot using pxctl", ctx.App.Key, vol), func() {
+				err = Inst().V.ValidateCreateSnapshotUsingPxctl(vol.ID)
+				if err != nil {
+					expect(err.Error()).To(contain(errPureFileSnapshotNotSupported.Error()))
+				}
+			})
+			Step(fmt.Sprintf("get %s app's volume: %s then create cloudsnap using pxctl", ctx.App.Key, vol), func() {
+				err = Inst().V.ValidateCreateCloudsnapUsingPxctl(vol.ID)
+				if err != nil {
+					expect(err.Error()).To(contain(errPureFileSnapshotNotSupported.Error()))
+				}
+			})
+		}
+		Step(fmt.Sprintf("validating groupsnap for using pxctl"), func() {
+			err = Inst().V.ValidateCreateGroupSnapshotUsingPxctl()
+			if err != nil {
+				expect(err.Error()).To(contain(errPureFileSnapshotNotSupported.Error()))
+			}
+		})
+	})
+}
+
+// ValidateResizeFBPVC is the ginkgo spec for validating resize of volumes
+func ValidateResizeFBPVC(ctx *scheduler.Context, errChan ...*chan error) {
```

Comment:
seems we don't validate the new size after resizing?

Reply:
this is calling k8s own update PVC api, so I think if it didn't succeed, errors would be returend, I feel this should be sufficient for this test, but I could add a TODO note for later to actually check the PV size again

Reply:
yeah but even it succeeded on k8s side, the PVC could be in state of pending expansion, it's safer to add a new size check in the test. could do it in the future.